### DISCUSSION TITLE
Air gapped Environment

### DISCUSSION THREAD

**Discussion by senthil13** [project-copacetic/copacetic#430] [General]
How does this tool work in an air-gapped environment?
Every system is not open to the entire internet. If it is running inside a closed network, then what URL do we need to open?

Thanks
Senthil

**Comment by sozercan** [comment]
@senthil13 you would need to open the urls of package repositories your containers are pulling the packages from. these are entirely depend on the containers and package managers. URLs can be official reposotiries or your local package mirrors.

If you are using a local mirror and patching a distroless image, you might need to substitute the tooling containers with your own using source policies to make sure they pull from your local mirror  https://project-copacetic.github.io/copacetic/website/faq#can-i-replace-the-package-repositories-in-the-image-with-my-own

**Reply by DWS-guy** [reply]
Is this still valid in the current version? I am also working on a closed network and would like to overwrite the default tooling image
